File: c/94_free_cash_237A.c
#include <stdio.h>

int main(void) {
    int n;
    if (scanf("%d", &n) != 1) return 0;
    int freq[1440] = {0};
    int ans = 1;
    for (int i = 0; i < n; ++i) {
        int h, m;
        scanf("%d %d", &h, &m);
        int t = h * 60 + m;
        ++freq[t];
        if (freq[t] > ans) ans = freq[t];
    }
    printf("%d\n", ans);
    return 0;
}

File: python/94_free_cash_237A.py
n = int(input())
freq = [0] * (24 * 60)
ans = 1
for _ in range(n):
    h, m = map(int, input().split())
    t = h * 60 + m
    freq[t] += 1
    ans = max(ans, freq[t])
print(ans)
